### ISSUE TITLE
<Tabs> is not compatible with Angular logic

### ISSUE THREAD

**Issue by csimpi** [NativeScript/nativescript-angular#2130]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
Newest versions:
https://play.nativescript.org/?template=play-ng&id=A7dRqF&v=3
 
**Describe the bug**

`<Tabs>` doesn't work with Angular directives at all. This is a huge lack of NS since I can't create Tabs dynamically :( Hardcoding tabs feels meaningless.
I've tried many versions but I'm getting error messages and blank screen only. 

If I set az Initial value it appears, but when I update the input variable of ngFor it doesn't update the tabs at all. 

`ngIf` doesn't help, async pipe doesn't help.
 

**To Reproduce**

https://play.nativescript.org/?template=play-ng&id=A7dRqF&v=3

**Expected behavior**

Possibility generate tabs with `ngFor` or any other way just be dynamic, not static.

**Sample project**

https://play.nativescript.org/?template=play-ng&id=A7dRqF&v=3

**Additional context**

**Comment by csimpi** [comment]
**UPDATE:**

After hours of struggling, I could find a way to create tabs based on an array of objects, but it works only if that array is static and present before `ngOnInit()`. 

Tabs should be able to handle the change in these variables also should be able to handle objects that are not present before `ngOnInit()`, like async responses, observables etc...

Now I have to block page/app loading until I get the list of the tabs that I want and if I want to change them I had to reload the whole page. Very bad UX in my opinion.

**Comment by NickIliev** [comment]
@csimpi actually you are using Angular structural directives to create the tabs which are causing the issue with the native component. The thing is that the Tabs component needs to have the exact same number of TabStripItem and TabContentItem when initialized. And using the ANgulr structural directive means that (due to the async nature of how those sub-components are created) there will be a moment when the Tabs is initialized but with the wrong number of required sub0-components.

SO in conclusion - **don't** use *ngFor and *ngIf or other Angular directives from the top abstraction to creates a native mobile component from the bottom abstraction (iOS & Android).

Instead, you could entirely create your Tabs, TabsStrip, and items from the code-behind.

**Comment by csimpi** [comment]
@NickIliev I don't understand. Why is there a moment when the number of tabs and items are different? If the input's item count is 0, 0 tabs have 0 items 0 === 0. When items have been added the number of items and tabs should appear. n === n.

This is the Angular version of NS and every time the answer to my questions is that the NS codebase does not really support the Angular way. You ask me don't use *ngIf and *ngFor which are the most essential functions of Angular, so I'm confused now. Can we expect a real Angular support/version or basically these are not going to work either in a long term?

**Comment by Psique** [comment]
Good afternoon from Barcelona,

We have found a solution to the stated issue of using *ngFor directive with {N} Tabs.

Check out this working example:

```
<Tabs>
    <TabStrip>
        <ng-container *ngFor="let tab of tabs">
            <TabStripItem title="{{ tab.title }}"> </TabStripItem>
        </ng-container>
    </TabStrip>

    <ng-container *ngFor="let tab of tabs">
        <TabContentItem>
            <Label text="{{ tab.content }}"></Label>
        </TabContentItem>
    </ng-container>
</Tabs>
```
